.gitlab-ci.yml

```diff
@@ -1,197 +1,152 @@
 stages:
   - test
-  - push-release
   - build
 variables:
   RUST_BACKTRACE: "1"
   RUSTFLAGS: ""
   CARGOFLAGS: ""
   CI_SERVER_NAME: "GitLab CI"
   LIBSSL: "libssl1.0.0 (>=1.0.0)"
-  CARGO_HOME: $CI_PROJECT_DIR/cargo
 cache:
-  key: "$CI_BUILD_STAGE-$CI_BUILD_REF_NAME"
+  key: "$CI_JOB_NAME"
   paths:
     - target/
     - cargo/
   untracked: true
-linux-stable:
+.list_of_refs: &list_of_refs #list of git refs for build and/or push, deploy
   stage: build
-  image: parity/rust:gitlab-ci
   only:
+    - stable
     - beta
+    - gitlab
     - tags
-    - stable
     - triggers
+.artifacts_rules: &artifacts_rules
+  artifacts:
+    when: on_success
+    expire_in: 1 mos
+    paths:
+    - artifacts/
+linux-stable:
+  <<: *list_of_refs
+  image: parity/rust:gitlab-ci
   script:
     - rustup default stable
     # ARGUMENTS: 1. BUILD_PLATFORM (target for binaries) 2. PLATFORM (target for cargo) 3. ARC (architecture) 4. & 5. CC & CXX flags 6. binary identifier
     - scripts/gitlab-build.sh x86_64-unknown-linux-gnu x86_64-unknown-linux-gnu amd64 gcc g++ ubuntu
   tags:
     - rust-stable
-  artifacts:
-    paths:
-    - parity.zip
-    name: "stable-x86_64-unknown-linux-gnu_parity"
+  <<: *artifacts_rules
 linux-stable-debian:
-  stage: build
+  <<: *list_of_refs
   image: parity/rust-debian:gitlab-ci
-  only:
-    - beta
-    - tags
-    - stable
-    - triggers
   script:
     - export LIBSSL="libssl1.1 (>=1.1.0)"
     - scripts/gitlab-build.sh x86_64-unknown-debian-gnu x86_64-unknown-linux-gnu amd64 gcc g++ debian
   tags:
     - rust-debian
-  artifacts:
-    paths:
-    - parity.zip
-    name: "stable-x86_64-unknown-debian-gnu_parity"
+  <<: *artifacts_rules
 linux-centos:
-  stage: build
+  <<: *list_of_refs
   image: parity/rust-centos:gitlab-ci
-  only:
-    - beta
-    - tags
-    - stable
-    - triggers
   script:
     - scripts/gitlab-build.sh x86_64-unknown-centos-gnu x86_64-unknown-linux-gnu x86_64 gcc g++ centos
+  retry: 2 #if version `GLIBC_2.18' not found
   tags:
     - rust-centos
-  artifacts:
-    paths:
-    - parity.zip
-    name: "x86_64-unknown-centos-gnu_parity"
+  <<: *artifacts_rules
 linux-i686:
-  stage: build
+  <<: *list_of_refs
   image: parity/rust-i686:gitlab-ci
-  only:
-    - beta
-    - tags
-    - stable
-    - triggers
   script:
     - scripts/gitlab-build.sh i686-unknown-linux-gnu i686-unknown-linux-gnu i386 gcc g++ ubuntu
   tags:
     - rust-i686
-  artifacts:
-    paths:
-      - parity.zip
-    name: "i686-unknown-linux-gnu"
+  <<: *artifacts_rules
 linux-armv7:
-  stage: build
+  <<: *list_of_refs
   image: parity/rust-armv7:gitlab-ci
-  only:
-    - beta
-    - tags
-    - stable
-    - triggers
   script:
     - scripts/gitlab-build.sh armv7-unknown-linux-gnueabihf armv7-unknown-linux-gnueabihf armhf arm-linux-gnueabihf-gcc arm-linux-gnueabihf-g++ ubuntu
   tags:
     - rust-arm
   artifacts:
     paths:
-    - parity.zip
+    - artifacts/
+    expire_in: 1 mos
```

Comment:
same here, I think would be better to keep fixed number of artifacts instead.

Reply:
no, we have a difference in quantity
https://gitlab.parity.io/parity/parity/-/jobs/80231/artifacts/browse/artifacts/ arm64 snap
https://gitlab.parity.io/parity/parity/-/jobs/80228/artifacts/browse/artifacts/ arm64 deb